analysis_driver/pipelines/demultiplexing.py

```diff
@@ -28,14 +28,13 @@ def _run(self):
         self.info('Input BCL folder: ' + self.input_dir)
         self.info('Fastq dir: ' + self.fastq_dir)
 
-        if not isdir(self.fastq_dir):
-            makedirs(self.fastq_dir)
+        makedirs(self.fastq_dir, exist_ok=True)
 
         # Send the information about the run to the rest API
         crawler = RunCrawler(self.dataset)
         crawler.send_data()
 
-        b = BCLValidator(self.job_dir, self.dataset)
+        b = BCLValidator(dataset=self.dataset)
```

Comment:
This worked because by coincidence, the Luigi parameters were getting correctly assigned as positional arguments to `self.previous_stages` (which we weren't using because we weren't using BCLValidator.run) and `self.dataset`